libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1092,60 +1099,76 @@ explainSpec _ s = s
 
 simplifySpec :: HasSpec fn a => Specification fn a -> Specification fn a
 simplifySpec spec = case spec of
-  SuspendedSpec x p -> explainSpec [show $ "Simplifying" /> pretty spec] $ computeSpecSimplified x (optimisePred p)
+  SuspendedSpec x p ->
+    let optP = optimisePred p
+     in fromGESpec
+          $ explain
+            [ show $ "Simplifying: " /> pretty spec
+            , show $ "optimisePred =>" /> pretty optP
+            ]
+          $ computeSpecSimplified x optP
   MemberSpec xs -> MemberSpec (nub xs)
   ErrorSpec es -> ErrorSpec es
   TypeSpec ts cant -> TypeSpec ts (nub cant)
   TrueSpec -> TrueSpec
 
 -- | Precondition: the `Pred fn` defines the `Var a`
 computeSpecSimplified ::
-  forall fn a. (HasSpec fn a, HasCallStack) => Var a -> Pred fn -> Specification fn a
-computeSpecSimplified x p = fromGESpec $ case p of
+  forall fn a. (HasSpec fn a, HasCallStack) => Var a -> Pred fn -> GE (Specification fn a)
```

Comment:
Comment that we run this in the GE  monad, in order to collect and display more accurate context when we report errors.